packages/shell-api/src/shell-internal-state.ts

```diff
@@ -35,6 +35,19 @@ export interface AutocompleteParameters {
   getCollectionCompletionsForCurrentDb: (collName: string) => Promise<string[]>;
 }
 
+export interface OnLoadParameters {
```

Comment:
Hm yeah … I’ll rename it, but it makes less sense that way to me – this inverses the control flow, and the return value of the function here is actually what is used to perform the actual evaluation instead of being the result itself

Reply:
Thanks! Mh probably depends on the point of view - from the method being implemented `onLoad` it looks strange to have parameters and in the `load` of Shell API these are used for setting context and evaluation...